src/ClientContext.ts

```diff
@@ -210,54 +165,47 @@ export class ClientContext {
   public async create<T>(
     body: T,
     path: string,
-    type: ResourceType,
-    id: string,
-    initialHeaders: CosmosHeaders,
-    options?: RequestOptions
+    resourceType: ResourceType,
+    resourceId: string,
+    options: RequestOptions
   ): Promise<Response<T & Resource>>;
 
   // But a few cases, like permissions, there is additional junk added to the response that isn't in system resource props
   public async create<T, U>(
```

Comment:
not related to this PR, but this overload looks unnecessary.

Reply:
Tracking this here now https://github.com/Azure/azure-cosmos-js/issues/282